Source/EventHorizon/Producer/ConsumerService.cs

```diff
@@ -93,13 +97,20 @@ public class ConsumerService : ConsumerBase
             arguments.Partition,
             arguments.PublicStream);
 
-        if (!TryGetConsent(arguments, out var consent, out var failureResponse))
+        var consentFound = TryGetConsent(arguments, out var consent, out var failureResponse);
```

Comment:
i think i'd check the config first, no need to do work about consents if it's going to be ignored

Reply:
It does make a difference, as the consent-id is passed to the consumer if it exists. It just does not force it to be there, and will default to returning "NoConsent" instead. It is also quite cheap to check :)